The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/15811611813/job/44563662716) [(plain)](https://github.com/rust-lang/rust/commit/549d0b67ea7943dea53838a6ce05493245327265/checks/44563662716/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
   --> library/core/src/marker/variance.rs:21:48
    |
15  | / macro_rules! phantom_type {
16  | |     ($(
17  | |         $(#[$attr:meta])*
18  | |         pub struct $name:ident <$t:ident> ($($inner:tt)*);
...   |
21  | |         pub struct $name<$t>($($inner)*) where <$t>: ?Sized;
    | |                                                ^^^^ currently unsupported
...   |
94  | |     )*};
95  | | }
    | |_- in this expansion of `phantom_type!`
...
178 | / phantom_type! {
179 | |     /// Zero-sized type used to mark a type parameter as covariant.
180 | |     ///
181 | |     /// Types used as part of the return value from a function are covariant. If the type is _also_
...   |
229 | |     pub struct PhantomInvariant<T>(PhantomData<fn(T) -> T>);
230 | | }
    | |_- in this macro invocation

error: expected one of `#`, `;`, lifetime, or type, found `:`
   --> library/core/src/marker/variance.rs:21:52
    |
15  | / macro_rules! phantom_type {
16  | |     ($(
17  | |         $(#[$attr:meta])*
18  | |         pub struct $name:ident <$t:ident> ($($inner:tt)*);
...   |
21  | |         pub struct $name<$t>($($inner)*) where <$t>: ?Sized;
    | |                                                    ^ expected one of `#`, `;`, lifetime, or type
...   |
94  | |     )*};
95  | | }
    | |_- in this expansion of `phantom_type!`
...
178 | / phantom_type! {
179 | |     /// Zero-sized type used to mark a type parameter as covariant.
180 | |     ///
181 | |     /// Types used as part of the return value from a function are covariant. If the type is _also_
...   |
229 | |     pub struct PhantomInvariant<T>(PhantomData<fn(T) -> T>);
230 | | }
    | |_- in this macro invocation

error[E0432]: unresolved imports `self::variance::PhantomContravariant`, `self::variance::PhantomCovariant`, `self::variance::PhantomInvariant`
  --> library/core/src/marker.rs:13:5
   |
13 |     PhantomContravariant, PhantomContravariantLifetime, PhantomCovariant, PhantomCovariantLifetime,
   |     ^^^^^^^^^^^^^^^^^^^^                                ^^^^^^^^^^^^^^^^ no `PhantomCovariant` in `marker::variance`
   |     |
   |     no `PhantomContravariant` in `marker::variance`
14 |     PhantomInvariant, PhantomInvariantLifetime, Variance, variance,
   |     ^^^^^^^^^^^^^^^^ no `PhantomInvariant` in `marker::variance`

error[E0412]: cannot find type `PhantomCovariant` in this scope
   --> library/core/src/marker/variance.rs:143:45
    |
143 |     pub struct PhantomCovariantLifetime<'a>(PhantomCovariant<&'a ()>);
    |                                             ^^^^^^^^^^^^^^^^ not found in this scope

error[E0412]: cannot find type `PhantomContravariant` in this scope
   --> library/core/src/marker/variance.rs:160:49
    |
160 |     pub struct PhantomContravariantLifetime<'a>(PhantomContravariant<&'a ()>);
    |                                                 ^^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0412]: cannot find type `PhantomInvariant` in this scope
   --> library/core/src/marker/variance.rs:175:45
    |
175 |     pub struct PhantomInvariantLifetime<'a>(PhantomInvariant<&'a ()>);
    |                                             ^^^^^^^^^^^^^^^^ not found in this scope

error: unused import: `crate::any::type_name`
 --> library/core/src/marker/variance.rs:4:5
  |
---

error[E0425]: cannot find function, tuple struct or tuple variant `PhantomCovariant` in this scope
   --> library/core/src/marker/variance.rs:143:45
    |
143 |     pub struct PhantomCovariantLifetime<'a>(PhantomCovariant<&'a ()>);
    |                                             ^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find function, tuple struct or tuple variant `PhantomContravariant` in this scope
   --> library/core/src/marker/variance.rs:160:49
    |
160 |     pub struct PhantomContravariantLifetime<'a>(PhantomContravariant<&'a ()>);
    |                                                 ^^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find function, tuple struct or tuple variant `PhantomInvariant` in this scope
   --> library/core/src/marker/variance.rs:175:45
    |
175 |     pub struct PhantomInvariantLifetime<'a>(PhantomInvariant<&'a ()>);
    |                                             ^^^^^^^^^^^^^^^^ not found in this scope

error: unused import: `Hasher`
 --> library/core/src/marker/variance.rs:7:25
  |
```

</details>

--- a/library/core/src/marker/variance.rs
+++ b/library/core/src/marker/variance.rs
@@ -18,7 +18,7 @@ macro_rules! phantom_type {
         pub struct $name:ident <$t:ident> ($($inner:tt)*);
     )*) => {$(
         $(#[$attr])*
-        pub struct $name<$t>($($inner)*) where T: ?Sized;
+        pub struct $name<$t>($($inner)*) where $t: ?Sized;
 
         impl<T> $name<T>
             where T: ?Sized

